pkg/verify/signature.go

```diff
@@ -269,3 +254,110 @@ func limitSubjects(statement *in_toto.Statement) error {
 	}
 	return nil
 }
+
+type multihasher struct {
+	hashfuncs []crypto.Hash
+	hashes    []hash.Hash
+}
+
+func newMultihasher(hashfuncs []crypto.Hash) *multihasher {
```

Comment:
```suggestion
func newMultihasher(hashfuncs []crypto.Hash) (*multihasher, error) {
```

Reply:
I rather be extra cautious here so we don't can fall into a trap where nothing is selected and that somehow can trigger a valid comparison. There is a check in `getHashFunctions`, but I rather have extra safety nets for this code, as subject matching is very important to get right. So I would prefer that each function verifies the input first.

Reply:
Added the return value and also wrote a table-based test for it.